src/components/AvatarMenu/AvatarMenu.types.ts

```diff
@@ -0,0 +1,27 @@
+import type { AvatarProps as MuiAvatarProps } from '@mui/material/Avatar';
+import type { MenuItemProps as MuiMenuItemProps } from '@mui/material/MenuItem';
+import type { MouseEventHandler, ReactNode } from 'react';
+
+export interface AvatarMenuItem {
+	label: string;
+	onClick: () => void;
+	icon?: ReactNode;
+	disabled?: boolean;
+	divider?: boolean;
+	'data-testid'?: string;
+	menuItemProps?: Omit<MuiMenuItemProps, 'onClick'>;
+}
+
+export interface PropsAvatarMenu {
+	avatarUrl: string;
+	username: string;
+	menuItems?: AvatarMenuItem[];
+	avatarProps?: Omit<MuiAvatarProps, 'src' | 'alt'>;
+	fullWidth?: boolean;
+	'data-testid'?: string;
+}
+
+export interface PropsStyledAvatarMenu {
+	$isOpen: boolean;
```

Comment:
PropsStyledAvatarMenu defines both $isOpen and $fullWidth as required booleans. In usage $fullWidth can be optional; consider making these props optional (or ensure you always pass both). Also styled props should be prefixed with $ and omitted from DOM via shouldForwardProp.

**Sugerencia:**
```typescript
Make these props optional if not always provided: export interface PropsStyledAvatarMenu { $isOpen?: boolean; $fullWidth?: boolean; } and ensure shouldForwardProp excludes them all.
```